### DISCUSSION TITLE
Option to remove extraneous page content

### DISCUSSION THREAD

**Discussion by sinewave** [coddingtonbear/obsidian-web#3] [Ideas]
Wow, just stumbled across this extension and it looks amazing!! Really loving the different template abilities and all the configurability.

One request I'd like to make after comparing the page content that's downloaded to something like MarkDownload, which is the extension I've been using, is the difference in the amount of content that's extracted.

Obsidian Web also seems to be capturing a lot of extraneous content such as page navigation links, comments, hidden markup, etc..

Perhaps an option to download a stripped down "Readability" version of the page?

**Comment by coddingtonbear** [comment]
Using a "readable" version should probably be the default/only option if somebody were to put that together into a Pull Request.  Really: it's probably trivial to make that work as long as however you get the "readable" version doesn't require a sandbox.

If somebody were to take this on before I get to this; the current process is:

1. Gather content from the active tab (https://github.com/coddingtonbear/obsidian-web/blob/master/src/popup.tsx#L114)
2. Run it through [turndown](https://www.npmjs.com/package/turndown) (https://github.com/coddingtonbear/obsidian-web/blob/master/src/popup.tsx#L118)

_Probably_ there are further steps that MarkDownload is taking; I'm just not sure exactly what they are having not looked into it in any significant depth.

**Comment by coddingtonbear** [comment]
Spent a few minutes on this this morning & found that [Markdownload](https://github.com/deathau/markdownload) is using [@mozilla/readability](https://github.com/mozilla/readability) for trimming content.

I've added a basic implementation of this in b61937cf0272355dd9ffe4c88c5d547474c0607a, but clearly markdownload is doing a little more work than just running the content through readability; so this won't create feature parity.  That all being said, this is probably good enough for at least my uses; so that's as far as I'm willing to carry this particular feature personally.
